internal/controllers/addon/addon_instance_reconciler.go

```diff
@@ -27,8 +28,10 @@ type addonInstanceReconciler struct {
 func (r *addonInstanceReconciler) Reconcile(ctx context.Context,
 	addon *addonsv1alpha1.Addon) (reconcile.Result, error) {
 	// Ensure the creation of the corresponding AddonInstance in .spec.install.olmOwnNamespace/.spec.install.olmAllNamespaces namespace
-	if err := r.ensureAddonInstance(ctx, addon); err != nil {
+	if requeueResult, err := r.ensureAddonInstance(ctx, addon); err != nil {
```

Comment:
Adding a `requeueResult` return is unnecessary here, we either get retry based on errors or get events when the addon instance is modified.